### PR TITLE
fix(alerts): Prevent form submit when adding team to project

### PR DESCRIPTION
**Pull request by scttcper**
Adding teams previously submitted the form, this updates the team to display correctly after adding and keeps you on the page.

Also, the close react-select logic wasn't working.

closes https://getsentry.atlassian.net/browse/WOR-142
fixes JAVASCRIPT-222Z

Kapture 2020-08-13 at 9 11 38

### CODE REVIEW

File: src/sentry/static/sentry/app/components/selectMembers/index.tsx [hunk 1]
```diff
@@ -104,9 +104,6 @@ class SelectMembers extends React.Component<Props, State> {
 
   unlisteners = [
     // See comments in `handleAddTeamToProject` for why we close the menu this way
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/selectMembers/index.tsx:106]: Does this comment still apply?

**Review by billyvg — COMMENTED** [review]
Have you tested the other views that use this component? I think the Ownership Rules also uses this

**Comment by scttcper** [comment]
@billyvg does seem to still work with ownership rules

File: src/sentry/static/sentry/app/components/selectMembers/index.tsx [hunk 2]
```diff
@@ -255,12 +253,23 @@ class SelectMembers extends React.Component<Props, State> {
       // wait for store to update before closing the menu. Otherwise, we'll have stale items in the select menu
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/selectMembers/index.tsx:253]: Does this still apply?

Comment (scttcper) [reply src/sentry/static/sentry/app/components/selectMembers/index.tsx]: comment no longer applies.

**Comment by scttcper** [comment]
ownership rules seems to be using another component, this seems to be shared with mail actions on the older issue actions
image

**Review by billyvg — APPROVED** [review]

**Pull request merged**